packages/ckb-indexer/src/collector.ts

```diff
@@ -41,197 +42,181 @@ export class CKBCellCollector implements BaseCellCollector {
       outputCapacityRange: undefined,
       bufferSize: undefined,
     };
-    this.queries = { ...defaultQuery, ...this.queries };
+    this.queries = (Array.isArray(queries) ? queries : [queries]).map(
+      (query) => ({ ...defaultQuery, ...query })
+    );
     this.validateQueryOption(this.queries);
     this.convertQueryOptionToSearchKey();
   }
 
-  public validateQueryOption(queries: CKBIndexerQueryOptions): void {
-    if (!queries.lock && (!queries.type || queries.type === "empty")) {
-      throw new Error("Either lock or type script must be provided!");
-    }
+  public validateQueryOption(queries: CKBIndexerQueryOptions[]): void {
+    queries.forEach((query) => {
```

Comment:
Maybe we don't need to modify this method

```js
this.queries.forEach(this.validateQueryOption.bind(this))
```

this will make the nesting depth less and make the code look cleaner